src/Microsoft.ML.Data/Utilities/StreamUtils.cs

```diff
@@ -174,5 +174,39 @@ private static string[] Expand(string pattern)
             return matchList.ToArray();
         }
 #endif
+
+        public static string ReadEntry(this TextReader sr)
+        {
+            string entry = string.Empty;
+
+            // get first bit
+            entry += sr.ReadLine();
+
+            // And get more lines until the number of quotes is even
+            while (GetNumberOf(entry, "\"") % 2 != 0 )
```

Comment:
Need to first check that that quoting is enabled, and need to check for escaped quotes.

Some tests:
* `True,"asdf,123` is perfectly ok without quoting enabled.
* `True,"hi\"asdf",123` has an escaped quote.

Reply:
Also need to not append to lines at the start of the file which begin with `\\` or `#`: 
```c#
if (entry[0] == '#' ||  entry[0] == '/' && entry[1] == '/') { ... }
```

Reply:
So try to repeat the logic here

if line starts with '#' or starts with '\\' => return line    # comments
if quote is disabled: => use the existing logic in TextLoader
else
if ( line has only odd# real quote ) => read next line until sum of real quote is even  and return line # real quote means `"` and it's not one of [ `""`, `\"` ` "`]
else return line
    

And BTW it seems the existing enable quote logic doesn't check if it's real quote or not. ( for example in lines: "blablabla""blablabla" will cause an error in `FetchNextField` function, but in Excel it's legal) Do we have a plan for adding support for that?